lua/rzls/handlers/providehtmldocumentcolor.lua

```diff
@@ -0,0 +1,39 @@
+local documentstore = require("rzls.documentstore")
+local razor = require("rzls.razor")
+local Log = require("rzls.log")
+
+local empty_response = {}
+
+---@param _err lsp.ResponseError
+---@param result lsp.DocumentDiagnosticParams
+---@param _ctx lsp.HandlerContext
+---@param _config table
+return function(_err, result, _ctx, _config)
+    assert(not _err, vim.inspect(_err))
+    local virtual_document =
+        documentstore.get_virtual_document(result.textDocument.uri, razor.language_kinds.html, "any")
```

Comment:
```suggestion
        documentstore.get_virtual_document(result.textDocument.uri, razor.language_kinds.html, result._razor_hostDocumentVersion)
```

We need to wait for the document store to be synced before we return results for this, so we need to pass target version

Reply:
this fixed the de-syncing issue I was seeing